tools/ec2recovery.py

```diff
@@ -0,0 +1,284 @@
+        print "\t0) Exit\n\n"
+        ami_num = input("  Desired AMI [0-%s]: " % sel_num)
+        if ami_num in range(0, len(result)):
+            return result[ami_num]
+
+
+def create_bdm(snaps):
+    bdm = []
+    for snap in snaps:
+        volume = {}
+        volume['DeviceName'] = snap[1]
+        volume['Ebs'] = {}
+        volume['Ebs']['SnapshotId'] = snap[2]
+        bdm.append(volume)
+    return bdm
+
+
+def create_tags(tags):
+    result = []
+    for i in range(len(tags)):
+        key = tags[i]['Key']
+        value = tags[i]['Value']
+        if key == 'Name':
+            value = "%s-restored" % value
+        if key == 'RestoreDate' or 'aws:' in key:
+            continue
+        result.append({
+            'Key': key,
+            'Value': value})
+    result.append({
+        'Key': 'RestoreDate',
+        'Value': datetime.now().strftime("%Y-%m-%d %H:%M:%S")})
+    return result
+
+
+def create_instance(session, dry_run, ami_id, instance, bdm, user_data):
+    # Create volumes from snapshots
+    ec2 = session.resource('ec2')
+    instances = ec2.create_instances(
+        DryRun=dry_run,
+        ImageId=ami_id,
+        MinCount=1,
+        MaxCount=1,
+        KeyName=instance['KeyName'],
+        SecurityGroupIds=pull_sg_ids(instance),
+        InstanceType=instance['InstanceType'],
+        SubnetId=instance['SubnetId'],
+        BlockDeviceMappings=bdm,
+        IamInstanceProfile=pull_iam_profile(instance),
+        UserData=user_data)
+    instance_id = instances[0].instance_id
+    restored = ec2.Instance(instance_id)
+    print "\n\tRestoring instance from snapshot..."
+    while restored.state['Name'] != 'running':
+        time.sleep(5)
+        restored = ec2.Instance(instance_id)
+    restored.create_tags(
+        DryRun=dry_run,
+        Tags=create_tags(instance['Tags']))
+    return instance_id
+
+
+def confirm_selection(snapshots, ami):
+    while True:
+        clear_screen()
+        print "\tVerify Restore Options\n\n"
+        print "\tAMI ID: %s\n" % ami
+        print "\tSnapshots:"
+        for snap in snapshots:
+            print "\t\tVolumeID: %s, VolumeName: %s, SnapshotID: %s\n" % (
+                snap[0], snap[1], snap[2])
+        response = raw_input("\nAre these values correct? [Y/N]: ")
+        if response.upper() == "Y" or response.upper() == "N":
+            return response
+
+
+def pull_iam_profile(instance):
+    iam = instance['IamInstanceProfile']
+    arn = iam['Arn']
+    name = arn.split('/')[1]
+    return {'Arn': arn}
+
+
+def main():
+    clear_screen()
+    parser = generate_parser()
+    options, args = parser.parse_args()
+    if not options.instance_id:
+        sys.exit(-1)
+
+    session = boto3.Session(
+        profile_name=options.profile,
+        region_name=options.region)
+    client = session.client("ec2")
+    instance = pull_instance(
+        options.dry_run,
+        client,
+        options.instance_id)
+    snapshots = pull_snapshots(
```

Comment:
instead of pulling snaps by volume given instance, you could just pull snaps direct given instance id since their tagged with it.